File: 2533-bitwise-xor-of-all-pairings/bitwise-xor-of-all-pairings.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
var xorAllNums = function(nums1, nums2) {
    let n = nums1.length, m = nums2.length;

    let ans = 0;
    if(m & 1) {
        for(let val of nums1)
            ans = ans ^ val;
    }

    if(n & 1) {
        for(let val of nums2)
            ans = ans ^ val;
    }

    return ans;
};

File: 2533-bitwise-xor-of-all-pairings/bitwise-xor-of-all-pairings.cpp
class Solution {
public:
    int xorAllNums(vector<int>& nums1, vector<int>& nums2) {
        int n  = nums1.size(), m = nums2.size();

        int ans = 0;

        for(int i = 0; i < n; i++) {
            if(m % 2) ans ^= nums1[i];
        }

        for(int i = 0; i < m; i++) {
            if(n % 2) ans ^= nums2[i];
        }

        return ans;
    }
};